app/assets/javascripts/plugins.js

```diff
@@ -23,12 +23,13 @@ window.Plugins = {
 
   // Its likely we dont need most of these context values. TBD.
   Interface IRuntimeContext {
-    name: string;         // Name of the plugin
-    url: string;          // Url from which the plugin was loaded
-    pluginId: string;     // Active record ID of the plugin scope id
-    authorData: string;   // The authored configuration for this instance
-    learnerData: string;  // The saved learner data for this instance
-    div: HTMLElement;     // reserved HTMLElement for the plugin output
+    name: string;               // Name of the plugin
+    url: string;                // Url from which the plugin was loaded
+    pluginId: string;           // Active record ID of the plugin scope id
+    authorData: string;         // The authored configuration for this instance
+    learnerData: string;        // The saved learner data for this instance
+    pluginUserStatePath: string // Where to save / load data from for this instance.
```

Comment:
I wouldn't really pass it to the plugin, so they can't mess with it. E.g. someone can write plugin posting requests manually and we won't be able to change paths, underlying implementation, etc. If we force plugin to use JS API, we have more flexibility.